### ISSUE TITLE
Predicted bounding boxes are consistently smaller than objects on AI-TOD

### ISSUE THREAD

**Issue by VickkyMama** [ZhuHaoranEIS/Point-Teacher#7]
Hi,

Thank you for releasing the paper and code.

I trained Point-Teacher using the official repository on the AI-TOD dataset with HBB annotations. During inference, I observed that the predicted bounding boxes are consistently smaller than the full object extent and appear to be very similar in size across different objects (almost like fixed-size inner boxes).

I would like to clarify whether this behavior is expected due to the point-supervision and pseudo-label generation strategy, or if it indicates an issue with my configuration or dataset preparation.

In addition, when I tried training and testing on SODA-A (OBB) and DOTA (HBB), the model produced zero detections. I would appreciate any guidance on whether additional configuration changes are required for DOTA, or if the method is not intended to generalize directly to that dataset.

Thank you for your time and for making the code publicly available.

<img width="1409" height="707" alt="Image" src="https://github.com/user-attachments/assets/bd123fd0-13bf-47cf-b3f2-a8dd66a9aebf" />

**Comment by ZhuHaoranEIS** [comment]
Thank you very much for your attention and insightful feedback!
Indeed, as you can imagine, MIL-based methods face significant challenges when applied to tiny objects—particularly in accurately capturing precise aspect ratio information. This is a capability that MIL tends to retain more effectively for larger objects, where richer spatial cues are available within the instance-level representations.
Consequently, in the context of small objects, MIL often resorts to relatively coarse scale estimation rather than fine-grained geometric regression. 

Regarding the phenomenon illustrated in your figure. A few practical adjustments could help: 
(1) slightly reducing the learning rate to improve convergence stability, 
(2) dampening the fluctuation of the regression loss, 
(3) revising the size distribution of synthetic objects.


Moreover, with respect to your comment on the DOTA dataset, you can try some hyperparameter configuration tuning:
shape_list = [[20, 20, 0.2, 0.5], [10, 20, 0.5, 0.5], [10, 30, 0.3, 0.7],
              [40, 20, 0.5, 0.5], [30, 20, 0.1, 0.8], [20, 50, 0.5, 0.5],
              [70, 64, 0.3, 0.6], [35, 60, 0.7, 0.4]]
fine_proposal_cfg=[
            dict(
                gen_mode='refine',
                gen_proposal_mode='fix_gen',
                cut_mode=None,
                shake_ratio=None,
                base_ratios=[1.0],
                min_scale=4,
                pos_iou_thr=0.3,
                neg_iou_thr=0.3,
                gen_num_neg=200),
            dict(
                gen_mode='refine',
                gen_proposal_mode='fix_gen',
                cut_mode=None,
                shake_ratio=None,
                base_ratios=[1.0],
                min_scale=0,
                pos_iou_thr=0.3,
                neg_iou_thr=0.3,
                gen_num_neg=200)
        ],
        fine_proposal_extensive_cfg=[
            dict(
                gen_mode='refine',
                gen_proposal_mode='fix_gen',
                cut_mode=None,
                shake_ratio=None,
                base_ratios=[1.0, 1.3, 1.5, 2.0, 0.8, 0.7, 0.5],
                min_scale=4,
                pos_iou_thr=0.3,
                neg_iou_thr=0.3,
                gen_num_neg=0),
            dict(
                gen_mode='refine',
                gen_proposal_mode='fix_gen',
                cut_mode=None,
                shake_ratio=None,
                base_ratios=[1.0, 1.3, 0.8],
                min_scale=4,
                pos_iou_thr=0.3,
                neg_iou_thr=0.3,
                gen_num_neg=0)
        ]),

Please feel free to share further thoughts or questions—I’d be glad to continue this discussion!

**Comment by VickkyMama** [comment]
Thank you for your quick response. I have a question regarding the **shape_list** used in the config/detector implementation.

From reading the paper and examining the code, it seems that shape_list is a handcrafted set of synthetic priors. Could you clarify how these values were derived? For example, do they represent typical object shapes per dataset (e.g., in AI-TODv2 HBB there are 6 entries), or were they chosen based on dataset statistics such as object size and aspect ratio distributions?

In the file fcos_p2b_teacher_student.py, the line below loads the shape_list. I would greatly appreciate any explanation of the rationale behind these values, as it would help me adapt the method to other datasets.

self.pattern, self.prior_size = load_basic_shape(shape_list)

Thank you!
